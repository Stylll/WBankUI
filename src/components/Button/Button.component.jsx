import React from 'react';
import PropTypes from 'prop-types';

import './Button.styles.scss';
import Loader from '../Loader/Loader.component';

const Button = (props) => {
  const {
    title, handleClick, disabled, customClassName,
    customStyles, buttonType, showLoader, dataTestId,
  } = props;

  const loaderStyles = {
    width: '10px',
    height: '10px',
  };

  return (
    <div className="btn-button">
      <button data-testid={dataTestId} type={buttonType} className={`button ${customClassName}`}
        style={customStyles} onClick={handleClick} disabled={disabled}
      >
        {showLoader ? <Loader customStyles={loaderStyles} /> : title}
      </button>
    </div>
  );
};

Button.propTypes = {
  handleClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  customStyles: PropTypes.object,
  customClassName: PropTypes.string,
  buttonType: PropTypes.string,
  disabled: PropTypes.bool,
  showLoader: PropTypes.bool,
  dataTestId: PropTypes.string,
};

Button.defaultProps = {
  handleClick: () => {},
  buttonType: 'submit',
  customClassName: '',
  disabled: false,
  showLoader: false,
  dataTestId: 'button',
};

export default Button;
