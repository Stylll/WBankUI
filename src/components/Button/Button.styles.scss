@import '../../partials/colors';

.btn-button {
  padding: 4px 2px;
  &>button {
    width: 140px;
    height: 40px;
    border-radius: 5px;
    border: 1px solid $primary-color;
    background-color: $white;
    display: flex;
    justify-content: center;
    align-items: center;
    color: $white;
    text-transform: capitalize;
    cursor: pointer;
    outline: none;
    color: $primary-color;
    font-size: 14px;
    font-weight: 700;
    background-image: linear-gradient($secondary-color, $secondary-color);
    background-size: 100% 0%;
    background-repeat:no-repeat;
    background-position: bottom;
    transition: all 0.2s ease-in-out;
    &:hover {
      color: $white !important;
      background-size: 100% 100%;
      border: none;
    }
    &:active {
      background-image: linear-gradient($primary-color, $primary-color);
      background-size: 100% 100%;
      border: none;
      color: $white;
    }
    &:disabled {
      color: $gray !important;
      border-color: $gray !important;
      cursor: default;
      .up-loader {
        border-color: $gray;
        border-top-color: $white;
      }
      &:hover {
        background-size: 100% 0%;
        border-color: $gray !important;
        border: 1px solid $gray;
        color: $gray !important;
      }
    }
  }
}
