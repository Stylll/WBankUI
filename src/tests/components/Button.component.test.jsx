import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import Button from '../../components/Button/Button.component';

describe('Button Component', () => {
  it('should render properly', () => {
    const { baseElement } = render(<Button
        title="Deposit"
        handleClick={() => {}}
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly when disabled', () => {
    const { baseElement } = render(<Button
        title="Deposit"
        handleClick={() => {}}
        disabled
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly when showLoader is true', () => {
    const { baseElement } = render(<Button
        title="Deposit"
        handleClick={() => {}}
        showLoader
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call handleClick method when button is clicked', () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(<Button
        title="Deposit"
        handleClick={handleClick}
        showLoader
        dataTestId="button-test"
    />);

    fireEvent.click(getByTestId('button-test'));
    expect(handleClick).toHaveBeenCalled();
  });
});
