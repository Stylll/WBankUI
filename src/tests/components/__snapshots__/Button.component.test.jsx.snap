// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render properly 1`] = `
<body>
  <div>
    <div
      class="btn-button"
    >
      <button
        class="button "
        data-testid="button"
        type="submit"
      >
        Deposit
      </button>
    </div>
  </div>
</body>
`;

exports[`Button Component should render properly when disabled 1`] = `
<body>
  <div>
    <div
      class="btn-button"
    >
      <button
        class="button "
        data-testid="button"
        disabled=""
        type="submit"
      >
        Deposit
      </button>
    </div>
  </div>
</body>
`;

exports[`Button Component should render properly when showLoader is true 1`] = `
<body>
  <div>
    <div
      class="btn-button"
    >
      <button
        class="button "
        data-testid="button"
        type="submit"
      >
        <div
          class="loader-container "
        >
          <div
            class="loader"
            style="width: 10px; height: 10px;"
          />
        </div>
      </button>
    </div>
  </div>
</body>
`;
